.github/workflows/branch-build.yaml

```
@@ -279,11 +279,11 @@ jobs:
   slackNotification:
     name: Slack Notification
     runs-on: ubuntu-latest
-    needs: [ update-cluster ]
+    needs: [ update-cluster, integration-tests ]
     if: always()
     steps:
       - name: Slack Notification
-        if: ${{ always() && needs.update-cluster.result != 'success' }}
+        if: ${{ needs.update-cluster.result == 'failure' || needs.integration-tests.result == 'failure' }}
```

👍 I think it makes sense to post Slack updates also on failed integration tests on main branch. 
But maybe we can use simply `${{ failure() }}` here?

https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#failure

```suggestion
        if: ${{ failure() }}
```